src/WingetCreateCLI/Commands/BaseCommand.cs

```diff
@@ -282,9 +283,15 @@ protected static async Task<string> DownloadPackageFile(string installerUrl)
 
                 Logger.ErrorLocalized(nameof(Resources.DownloadFile_Error));
 
-                if (e is HttpRequestException exception)
+                if (e is HttpRequestException httpRequestException)
                 {
-                    Logger.ErrorLocalized(nameof(Resources.HttpResponseUnsuccessful_Error), exception.StatusCode);
+                    Logger.ErrorLocalized(nameof(Resources.HttpResponseUnsuccessful_Error), httpRequestException.StatusCode);
+                    return null;
+                }
+
+                if (e is DownloadSizeExceededException downloadSizeExceededException)
+                {
+                    Logger.ErrorLocalized(nameof(Resources.DownloadFileExceedsMaxSize_Error), $"{downloadSizeExceededException.MaxDownloadSize / 1024 / 1024}");
```

Comment:
Consider indicating the expected Units on the variable name. Ie MaxDownloadSizeInWhatever

Reply:
Max download size is nullable. C# i believe that will force the entire arithmetic to null.  should the download size not be optional?

Reply:
This exception will only ever be thrown if MaxDownloadSize is not null (see PackageParser.cs)

Reply:
Changed variable name to include expected units. I also changed the type of MaxDownloadSize to not be nullable so that the arithmetic doesn't compute to null. I believe that this is okay since DownloadSizeExceededException will only be called if a maxdownloadsize is properly defined.